File: python/blind75/arrays/two-sum.py
class Solution:
  def twoSum(self, nums, target):
    print('in twoSum')
    print(nums)
    print(target)
    hm = {}
    
    for i, n in enumerate(nums):
      diff = target - n
      if diff in hm:
        return [hm[diff], i]
      hm[n] = i
    
  def twoSumBad(self, nums, target):
    for i, n in enumerate(nums):
      for j, m in enumerate(nums):
        if n + m == target:
          print(n, m)
          return [i, j]
    return False


nums = [7,8,2,4]
target = 6

sol = Solution()

ans1 = sol.twoSum(nums, target)
print(ans1)
ans = sol.twoSumBad(nums, target)
print(ans)

File: node/twoSum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
 var twoSum = function(nums, target) {
  const hm ={};
  for (const [i, n] of nums.entries()) {
    const diff = target - n
    console.log(hm[diff] !== undefined)
    console.log(hm[diff])
    if(hm[diff] !== undefined) {
      console.log(diff)
      console.log([hm[diff], i])
      return [hm[diff], i]
    }
    // hm.set(n,i)
    hm[n] = i
  }
};

const nums = [2,7,11,15]
const target = 9

const ans = twoSum(nums, target)
console.log(ans)
